### PR TITLE
Download chromedriver binary for Mac M1

### PR DESCRIPTION
**Pull request by johnathafelix**
I noticed that using Node.js emulated with Rosetta made the script download the Intel version of the chromedriver instead of the Apple Silicon one. 

I've changed it the platform check so it can see what's the real processor architecture and download the chromedriver accordingly. 

I run browsers without emulation but not Node.js since some packages are still not compatible.

# We are in need of someone to test this

You need to be on an M1 Mac, this is how you do it:

Clone this PR:

````bash
git clone https://github.com/johnathafelix/node-chromedriver.git
cd node-chromedriver
npm install
````

Then, check if you can run Chromedriver:

````bash
./bin/chromedriver
````

If it shows output correctly, then it worked. This is the expected output (somewhat similar is ok):

````
Starting ChromeDriver 97.0.4692.71 (adefa7837d02a07a604c1e6eff0b3a09422ab88d-refs/branch-heads/4692@{#1247}) on port 9515
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
````

If it works, please comment bellow.

### CODE REVIEW

**Comment by johnathafelix** [comment]
A link for reference: https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment#3616845

Basically, `process.arch` returns `x86_64` if Node.js is running under Rosetta.

**Comment by giggio** [comment]
@johnathafelix It looks good. I don't have an M1 Mac to test, so I need somebody else to validate.
I added only one comment, please review it.
I was not aware that the Chrome team was now releasing `mac64_m1` binary. That is new, and your PR is welcome.

File: install.js [hunk 1]
```diff
@@ -330,6 +333,32 @@ function fixFilePermissions() {
...
+
+  if (process.arch === 'x64') {
+    return 'mac64';
+  }
+
+  return null;
+}
+
+function isEmulatedRosettaEnvironment() {
+  const archName = child_process.spawnSync('uname', ['-m']).stdout.toString().trim();
```
[the remark this answers is not in the document]
Comment (johnathafelix) [reply install.js:349]: Hi @giggio! In my tests, if I'm running Node.js under Rosetta 2, `process.arch` will return `x86_64` and will return `arm64` if running natively on Apple Silicon. 

So the only way I've found to get the real architecture of the processor is by running `uname -m` and then `sysctl -in sysctl.proc_translated`. 

Let me know if it's not clear 😃

**Comment by johnathafelix** [comment]
Thanks @giggio! Let's wait until someone else with a M1 Mac validates this.

File: install.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (giggio) [reply install.js:349]: Ok, I understand. So, what if we make the function like this:

```js
function isEmulatedRosettaEnvironment() {
  const processTranslated = child_process.spawnSync('sysctl', ['-in', 'sysctl.proc_translated'])
      .stdout.toString()
      .trim();
  return processTranslated === '1';
}
```

And then call it only from:

```js
  if (process.arch === 'x64') {
    if (isEmulatedRosettaEnvironment())
      return 'mac64_m1';
    return 'mac64';
  }
```

Would that work? If yes, we have one less process spawning.

[the remark this answers is not in the document]
Comment (giggio) [reply install.js:349]: Hi @johnathafelix! When do you expect to update this? I'd like to ask a friend to verify it after um change it.

**Comment by igor9silva** [comment]
This is the output I have, seems to be working properly:
image
image

**Comment by rodrigograudo** [comment]
Captura de Tela 2022-01-31 às 12 55 04

Tested on M1 Pro

**Comment by ricmello** [comment]
It seems to be working properly here too:

image

**Comment by samuelramox** [comment]
It's working here:
Screenshot 2022-01-31 às 14 15 18

But you might need to update a package with vulnerability: 
Screenshot 2022-01-31 às 14 26 40

**Comment by RodrigoRVieira** [comment]
Well... It seems some folks already replied, but... here it goes!

image

**Comment by nailsonlinux** [comment]
Here's my output and system specs.

Screen Shot 2022-02-01 at 07 40 43

Screen Shot 2022-02-01 at 07 41 09

**Comment by giggio** [comment]
Great, I merged it! Thanks for everyone that helped testing it!
I'll release a new version to NPM in a moment.
Thanks for the great PR @johnathafelix!

**Comment by giggio** [comment]
@samuelramox this branch was a little behind, the latest on `main` already has the vulnerabilities fixed. Thanks for reporting it.

**Comment by giggio** [comment]
Published as `97.0.4`. :shipit: 🚀

**Pull request merged**